spark/datadog_checks/spark/constants.py

```diff
@@ -37,7 +37,7 @@
 
 # Extract the application name and the dd metric name from the structured streams metrics.
 STRUCTURED_STREAMS_METRICS_REGEX = re.compile(
-    r"^[\w-]+\.driver\.spark\.streaming\.(?P<query_name>[\w-]+)\.(?P<metric_name>[\w-]+)$"
+    r"^[\w-]+\.driver\.spark\.streaming\.(?P<query_name>[\w/\\|!.,@-]+)\.(?P<metric_name>[\w-]+)$"
```

Comment:
Is there any documentation wrt which characters are allowed in query names? If there are no limitations, it may be worth simply allowing everything here:

```suggestion
    r"^[\w-]+\.driver\.spark\.streaming\.(?P<query_name>\S+)\.(?P<metric_name>[\w-]+)$"
```

Or even:

```suggestion
    r"^[\w-]+\.driver\.spark\.streaming\.(?P<query_name>.+)\.(?P<metric_name>[\w-]+)$"
```

Reply:
Honestly, I didn't find any docs talking about this topic.
From the spark's actual implementation, it seems that there is not any kind of limitation on streaming query names.

I think we could allow everything as you suggested. Thanks for your feedback